# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1315081228 25200
#      Sat Sep 03 13:20:28 2011 -0700
# Node ID 386024135ce6f8a44f3b07e2a13c2345e31510f2
# Parent  0b789a03bde1e8c53baa393cd5758538572b1f28
codesprint: Add test for onCleanup()

* onCleanup.cc: Add test for onCleanup()

diff --git a/src/DLD-FUNCTIONS/onCleanup.cc b/src/DLD-FUNCTIONS/onCleanup.cc
--- a/src/DLD-FUNCTIONS/onCleanup.cc
+++ b/src/DLD-FUNCTIONS/onCleanup.cc
@@ -269,8 +269,24 @@ For similar functionality @xref{The @cod
 
       retval = new octave_oncleanup (args(0));
     }
   else
     print_usage ();
 
   return retval;
 }
+
+/*
+
+%!test
+%! old_wstate = warning ("query");
+%! unwind_protect
+%!   trigger = onCleanup (@() warning ("on", "__MY_WARNING__"));
+%!   warning ("off", "__MY_WARNING__");
+%!   assert ((warning ("query", "__MY_WARNING__")).state, "off");
+%!   clear trigger
+%!   assert ((warning ("query", "__MY_WARNING__")).state, "on");
+%! unwind_protect_cleanup
+%!   warning (old_wstate);
+%! end_unwind_protect
+
+*/
